client/signup/steps/website-content/index.tsx

```diff
@@ -84,19 +86,30 @@ function WebsiteContentStep( {
 	const translatedPageTitles = useTranslatedPageTitles();
 
 	useEffect( () => {
-		if ( websiteContent.pages.length > 0 ) {
+		if (
+			websiteContent.pages.length &&
+			siteIdForSavedWebsiteContent &&
+			siteIdForSavedWebsiteContent === siteId
```

Comment:
I think this condition should be moved inside the reducer. And we should load existing state (previous state) only when there is no mismatch in the site IDs.

Reply:
Good point! This makes sense to me. I have moved the logic to the reducer and updated the tests.